TopMovies/Core/State/CategoriesPaginationState.swift

```diff
@@ -0,0 +1,84 @@
+//
+//  CategoriesPaginationState.swift
+//  TopMovies
+//
+//  Created by Macbook Pro  on 19.02.2021.
+//
+
+import ReSwift
+
+struct CategoriesPaginationState: StateType {
+  typealias CategoryState = PaginatedOperationState<Movie.ID, Int>
+  let paginated: [MovieCategory.ID: CategoryState]
+}
+
+extension CategoriesPaginationState.CategoryState: Defaultable {
+  typealias Value = CategoriesPaginationState.CategoryState
+  static var defaultValue: Value {
+    .init(list: [], reload: .initial, loadMore: .initial, pageInfo: .next(2))
+  }
+}
+// MARK: - Reducer -
+extension CategoriesPaginationState {
+  static func reduce(action: Action, state: CategoriesPaginationState) -> CategoriesPaginationState {
+    var newPaginated = state.paginated
+    switch action {
+    // MARK: - Reduce MovieCategoriesAction
+    case let completedCategories as CompletedMovieCategoriesAction:
+      return CategoriesPaginationState(
+        paginated: completedCategories.relational
+          .mapValues { CategoryState.init(list: $0,
+                                          reload: CategoryState.defaultValue.reload,
+                                          loadMore: CategoryState.defaultValue.loadMore,
+                                          pageInfo: CategoryState.defaultValue.pageInfo) })
+    // MARK: - Reduce MoviesListActions
+    case let requestedMoviesList as RequestedMoviesListAction:
+      let categoryId = requestedMoviesList.categoryId
+      newPaginated
+        .updateValue(.init(list: (state.paginated[categoryId] ?? .defaultValue) .list,
+                           reload: requestedMoviesList.requestType == .reload ?
+                            .requested : .initial,
```

Comment:
What if both reload and loadMore are requested? You will override one after 2nd action arrives since you set `.initial` instead of current value `state.paginated[categoryId] ?? .defaultValue).reload`.